engine/src/gldrv/winsys.cpp

```diff
@@ -265,136 +436,136 @@ static bool setup_sdl_video_mode(int *argc, char **argv) {
-                                0, 0, video_flags);
-    }
-
-    if(!window) {
-        VS_LOG_FLUSH_EXIT(fatal, "No window", 1);
-    }
+    renderer = nullptr;
 
-    if(screen_number > 0) {
-        // Get bounds of the secondary monitor
-        SDL_Rect displayBounds;
-        if (SDL_GetDisplayBounds(screen_number, &displayBounds) != 0) {
-            const std::string error_message = (boost::format("Failed to get display bounds: %1%") % SDL_GetError()).str();
-            VS_LOG_AND_FLUSH(error, error_message);
+    if (configuration().graphics.gl_accelerated_visual) {
+        result = try_creating_window_and_renderer("Vega Strike", width, height, "opengl", video_flags, &window, &renderer);
+        if (result == false) {
+            VS_LOG_AND_FLUSH(serious_warning, (boost::format("Creating window and renderer failed with error: %1%") % SDL_GetError()));
+            VS_LOG_AND_FLUSH(serious_warning, "Please make sure a graphics card driver is installed and functioning properly.");
+            SDL_ClearError();
+            (const_cast<vega_config::Configuration &>(configuration())).graphics.gl_accelerated_visual = false;
 
-            // Fallback to primary monitor
-            SDL_GetDisplayBounds(0, &displayBounds);
+            result = try_creating_window_and_renderer("Vega Strike", width, height, "software", video_flags, &window, &renderer);
+            if (result == false) {
+                VS_LOG_FLUSH_EXIT(fatal, (boost::format("Creating window and renderer failed even with software rendering! Error message: %1%") % SDL_GetError()), 1);
+            }
+        }
+    } else {
+        result = try_creating_window_and_renderer("Vega Strike", width, height, "software", video_flags, &window, &renderer);
+        if (result == false) {
+            VS_LOG_FLUSH_EXIT(fatal, (boost::format("Creating window and renderer failed even with software rendering! Error message: %1%") % SDL_GetError()), 1);
         }
-
-        // Move to secondary monitor
-        SDL_SetWindowPosition(window, displayBounds.x, displayBounds.y);
     }
-
-    if (configuration().graphics.full_screen) {
-        SDL_SetWindowFullscreen(window, SDL_WINDOW_FULLSCREEN_DESKTOP);
+    if (!SDL_SetWindowSize(window, width, height)) {
+        VS_LOG_AND_FLUSH(error, (boost::format("%1%: SDL_SetWindowSize(...) encountered an error: %2%") % __FUNCTION__ % SDL_GetError()));
+        SDL_ClearError();
     }
 
-    if (SDL_SetHint(SDL_HINT_RENDER_DRIVER, "opengl")) {
-        VS_LOG_AND_FLUSH(important_info, "SDL_SetHint(SDL_HINT_RENDER_DRIVER, ...) succeeded");
-    } else {
-        VS_LOG_AND_FLUSH(error, (boost::format("SDL_SetHint(SDL_HINT_RENDER_DRIVER, ...) failed. Error: %1%") % SDL_GetError()));
+    if (!SDL_GetCurrentRenderOutputSize(renderer, &native_resolution_x, &native_resolution_y)) {
+        VS_LOG_AND_FLUSH(error, (boost::format("%1%: SDL_GetCurrentRenderOutputSize(...) encountered an error: %2%") % __FUNCTION__ % SDL_GetError()));
```

Comment:
At a minimum, we should wrap these VS_LOG_AND_FLUSH(SDL_GetError) and SDL_ClearError in a function.

Reply:
Done